Interface.C

```diff
@@ -1,24 +1,37 @@
 #include "Interface.H"
 #include "Utilities.H"
+#include "faceTriangulation.H"
+
 
 using namespace Foam;
 
 preciceAdapter::Interface::Interface
 (
-    precice::SolverInterface & precice,
-    const fvMesh& mesh,
-    std::string meshName,
-    std::string locationsType,
-    std::vector<std::string> patchNames
-)
-:
-precice_(precice),
-meshName_(meshName),
-locationsType_(locationsType),
-patchNames_(patchNames)
+        precice::SolverInterface & precice,
+        const fvMesh& mesh,
+        std::string meshName,
+        std::string locationsType,
+        std::vector<std::string> patchNames
+        )
+    :
+      precice_(precice),
+      meshName_(meshName),
+      locationsType_(locationsType),
+      patchNames_(patchNames)
 {
     // Get the meshID from preCICE
-    meshID_ = precice_.getMeshID(meshName_);
+    //TODO: Don't use the suffixes in case of "Centers" or "Nodes"
+    if(locationsType_ == "faceTriangles" || locationsType_ == "faceCenters" || locationsType_ == "faceCentres"){
```

Comment:
Why do we need this?

Reply:
This has to do with the way the faceTriangles are specified by the user. In the yaml file, you just need to pass in `faceTriangles`. But in the background (and also in the `precice-config.xml`, we need to define two meshes: On for writing and one for reading. This is quite nasty for the user and we should think about another solution for the long term.

Reply:
This is outdated